app/scripts/controllers/provider-approval.js

```diff
@@ -48,7 +49,7 @@ class ProviderApprovalController extends SafeEventEmitter {
       if (approved) {
         res.result = [this.preferencesController.getSelectedAddress()]
       } else {
-        throw new Error('User denied account authorization')
+        rpcErrors.eth.userRejectedRequest('User denied account authorization.')
```

Comment:
Should this error be thrown? Or returned? I suspect `throw` would be appropriate

Reply:
Omitting `throw` was a mistake. I pushed a fix.